spec/validators/special_character_validator_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+RSpec.describe SpecialCharacterValidator do
+
+  let(:model) { ModelClass.new }
+  let(:valid_attributes) do
+    {
+      first_name: 'name',
+      last_name: 'last',
+      address_building: '12',
+      address_street: 'street name',
+      address_locality: 'local',
+      address_county: 'county',
+    }
+  end
+  let(:valid_attributes_with_comma) do
+    {
+      first_name: 'name',
+      last_name: 'last',
+      address_building: '12,,,,,',
+      address_street: 'street name',
+      address_locality: 'local',
+      address_county: 'county',
+    }
+  end
+  let(:invalid_attributes) do
+    {
+      first_name: 'qwe<',
+      last_name: '&*()',
+      address_building: '&*(',
+      address_street: ')(*',
+      address_locality: '*()',
+      address_county: '*()',
+    }
+  end
+  class ModelClass < ActiveRecord::Base
+    establish_connection adapter: :nulldb,
+                         schema: 'config/nulldb_schema.rb'
+
+    attribute :first_name,                :string
+    attribute :last_name,                :string
+    attribute :address_building,         :string
+    attribute :address_street,           :string
+    attribute :address_locality,         :string
+    attribute :address_county,           :string
+
+    validates :first_name, :last_name, special_character: true
+    validates :address_street, :address_locality, :address_county, special_character: true
```

Comment:
Why have you got so many fields setup ?

Surely you just need 2 - one with special_character: true and another with special_character: { comma: true } ?